lib/azure/armrest/storage/managed_storage_helper.rb

```diff
@@ -0,0 +1,123 @@
+  #
+  def get_blob_raw(disk_name, resource_group = configuration.resource_group, options = {})
+    validate_resource_group(resource_group)
+
+    post_options = {
+      :access            => 'read',                    # Must be 'read'
+      :durationInSeconds => options[:duration] || 3600 # 1 hour default
+    }
+
+    # This call will give us an operations URL in the headers.
+    initial_url = build_url(resource_group, disk_name, 'BeginGetAccess')
+    response    = rest_post(initial_url, post_options.to_json)
+    begin
+      headers = Azure::Armrest::ResponseHeaders.new(response.headers)
+
+      # Using the URL returned from the above call, make another call that
+      # will return the URL + SAS token.
+      op_url = headers.try(:azure_asyncoperation) || headers.location
+
+      unless op_url
+        msg = "Unable to find an operations URL for #{disk_name}/#{resource_group}"
+        raise Azure::Armrest::NotFoundException.new(response.code, msg, response.body)
+      end
+
+      # Dig the URL + SAS token URL out of the response
+      response = rest_get(op_url)
+      body     = Azure::Armrest::ResponseBody.new(response.body)
+      sas_url  = body.try(:properties).try(:output).try(:access_sas)
+
+      unless sas_url
+        msg = "Unable to find an SAS URL for #{disk_name}/#{resource_group}"
+        raise Azure::Armrest::NotFoundException.new(response.code, msg, response.body)
+      end
+
+      # The same restrictions that apply to the StorageAccont method also apply here.
+      range = options[:range] if options[:range]
+      range ||= options[:start_byte]..options[:end_byte] if options[:start_byte] && options[:end_byte]
+      range ||= options[:start_byte]..options[:start_byte] + options[:length] - 1 if options[:start_byte] && options[:length]
+
+      range_str = range ? "bytes=#{range.min}-#{range.max}" : nil
+
+      unless range_str || options[:entire_image]
+        raise ArgumentError, "must specify byte range or :entire_image flag"
+      end
+
+      headers = {}
+      headers['x-ms-range'] = range_str if range_str
+
+      # Need to make a raw call since we need to explicitly pass headers,
+      # but without encoding the URL or passing our configuration token.
+      max_retries = 5
+      retries     = 0
+      begin
+        RestClient::Request.execute(
+          :method      => :get,
+          :url         => sas_url,
+          :headers     => headers,
+          :proxy       => configuration.proxy,
+          :ssl_version => configuration.ssl_version,
+          :ssl_verify  => configuration.ssl_verify
+        )
+      rescue RestClient::Exception, Azure::Armrest::ForbiddenException => err
+        retries += 1
+        raise err unless retries < max_retries
+        log("get_blob_raw: 403 Forbidden received on read - retry number #{retries}")
+        retry
+      end
+    ensure
+      end_url = build_url(resource_group, disk_name, 'EndGetAccess')
```

Comment:
@djberg96 I'm not sure what you'd solve with that.  The retry is only for the failure of the actual read of the disk.  If anything prior to that (for instance the 'BeginGetAccess' call) fails for any reason we will hit the ensure clause but no retries will happen.